Sorting Algorithms/Merge Sort Algorithm/Java/MergeSort.java

```diff
@@ -0,0 +1,81 @@
+import java.util.Scanner;
+
+/* ===== ===== =====
+
+Theory of Programming
+
+Merge Sort Algorithm
+http://theoryofprogramming.com/2015/01/29/merge-sort-algorithm/
+GitHub - https://github.com/VamsiSangam/theoryofprogramming
+Code Contributor - Aditya Aggarwal (github.com/the-cybersapien, cybersapien.xyz)
+
+===== ===== ===== */
+public class MergeSort
+{
+
+    public static void main(String[] args){
+        //Initialize the Scanner Object to take input
+        Scanner in = new Scanner(System.in);
+        //Get the Size of the Array
+        System.out.println("Enter the size of the array:");
+        int size = in.nextInt();
+        //Initialize the Array for sorting the elements
+        int[] array = new int[size];
+        //Get the data from user
+        System.out.println("Enter the elements of the array: ");
+        for (int i = 0; i < size; i++) {
+            array[i] = in.nextInt();
+        }
+        //Merge the Array
+        mergeSort(array,0, size-1);
+        //Print the final answer to the User:
+        System.out.println("Sorted Array: ");
+        for (int x: array){
+            System.out.print(x + " ");
+        }
+        System.out.println();
+    }
+
+    private static void mergeSort(int array[], int low, int high){
+        //Check if low is smaller than higher
+        // this is the exit condition for the recursive solution
+        if (low < high){
+            //get the index of the element in the middle
+            int mid = low + (high-low)/2;
```

Comment:
Requesting another minor change. I requested for spaces on either side of every binary operator in the whole program not just for the lines mentioned in the previous review.